src/ranged.cpp

```diff
@@ -929,33 +991,46 @@ static int do_aim( player &p, const std::vector<Creature *> &t, int cur_target,
     return cur_target;
 }
 
+double Creature::projectile_attack_chance( double dispersion, double range, double accuracy ) const {
+    // This is essentially the inverse of what Creature::projectile_attack() does.
```

Comment:
sounds good. I think if you do this here and extend the other tests as discussed we should have fairly comprehensive coverage